src/main.tsx

```diff
@@ -2,9 +2,20 @@ import React from 'react'
 import ReactDOM from 'react-dom/client'
 import App from './App.tsx'
 import './App.css'
+import {configureStore} from "@reduxjs/toolkit"
+import { Provider } from 'react-redux'
+import reducer from './features/counter.tsx'
+
+const store = configureStore({
+  reducer:{
+    counter : reducer
```

Comment:
May you change the name of this **reducer** because it is hard to understand 
```suggestion
    counter : CounterReducer
```
also consider changing it in the import